frontend/src/_components/MultiSelect.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { FilterPreview } from '@/_components';
+import PropTypes from 'prop-types';
+import Select, { fuzzySearch } from 'react-select-search';
+import '@/_styles/widgets/multi-select.scss';
+
+function MultiSelect({
+  onSelect,
+  onSearch,
+  selectedValues,
+  onReset,
+  placeholder = 'Select',
+  options,
+  isLoading,
+  className,
+  searchLabel,
+}) {
+  const [searchText, setSearchText] = useState('');
+  const [filteredOptions, setOptions] = useState([]);
+
+  useEffect(() => {
+    options && setOptions(filterOptions(options));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options, selectedValues]);
+
+  const searchFunction = useCallback(
+    async (query) => {
+      setSearchText(query);
+      if (!query) {
+        return [];
+      }
+      const options = await onSearch(query);
+      return filterOptions(options);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [setSearchText, onSearch, selectedValues]
+  );
+
+  const filterOptions = useCallback(
+    (options) => {
+      if (selectedValues) {
+        return options?.filter((data) => !selectedValues?.some((selected) => selected.value === data.value));
+      }
+      return options;
+    },
+    [selectedValues]
+  );
+
+  return (
+    <div className="tj-ms tj-ms-count">
+      <FilterPreview
+        text={(selectedValues?.length?.toString() || '0') + ' selected'}
+        onClose={selectedValues?.length ? onReset : undefined}
+      />
+      <Select
+        className={className}
+        getOptions={onSearch ? searchFunction : undefined}
+        options={onSearch ? [] : filteredOptions}
+        closeOnSelect={false}
+        search={true}
+        multiple
+        value={{ name: '' }}
+        onChange={(id, value) => onSelect([...(selectedValues ? selectedValues : []), ...value])}
```

Comment:
If possible, we can add an empty array as a default/fallback value (`selectedValues=[]`) in the props to avoid undefined check on selectedValues everywhere 

```suggestion
        onChange={(id, value) => onSelect([...selectedValues, ...value])}
```